frontend/model/captureLogs.js

```diff
@@ -0,0 +1,250 @@
+  if (!isConsoleOveridden) {
+    // avoid duplicated captures, in case captureLogsStart gets
+    // called multiple times. (ex: login twice in the same visit)
+    isConsoleOveridden = true
+
+    // NOTE: Find a way to capture logs without messing up with log file location.
+    // console.log() doesnt include stack trace, so when logged, we can't access
+    // where the log came from (file name), which difficults debugging if needed.
+    window.console = new Proxy(window.console, {
+      get: (obj, type) => (...args) => {
+        if (appLogsFilter.includes(type)) {
+          obj[type](...args)
+          if (isCapturing) {
+            captureLog(type, ...args)
+            verifyLogsSize()
+          }
+        }
+      }
+    })
+  }
+
+  // Set a new visit or session - useful to understand logs through time.
+  // NEW_SESSION -> The user opened a new browser or tab.
+  // NEW_VISIT -> The user comes from an ongoing session (refresh or login)
+  const isNewSession = !sessionStorage.getItem('NEW_SESSION')
+  if (isNewSession) { sessionStorage.setItem('NEW_SESSION', 1) }
+  console.log(isNewSession ? 'NEW_SESSION' : 'NEW_VISIT', 'Starting to capture logs of type:', appLogsFilter)
+}
+
+export function captureLogsPause () {
+  captureLog('log', 'PAUSE_LOGS for:', username)
+  isCapturing = false
+  username = null
+  sbp('okTurtles.events/off', SET_APP_LOGS_FILTER)
+}
+
+export const getLog = giLSget
+
+// Util to download all stored logs so far
+export function downloadLogs (elLink) {
+  const filename = 'gi_logs.json'
+  const appLogsArr = []
+  let prevEntry = giLSget('lastEntry')
+  do {
+    const log = giLSget(`${prevEntry}`)
+    if (!log) break
+    prevEntry = JSON.parse(log).prev
+    appLogsArr.push(log)
+  } while (prevEntry)
+  appLogsArr.reverse() // chronological order (oldest to most recent)
+
+  const file = new Blob([JSON.stringify({
+    // Add instructions in case the user opens the file.
+    '_instructions': 'GROUP INCOME - Application Logs - Attach this file when reporting an issue: https://github.com/okTurtles/group-income-simple/issues',
+    'ua': navigator.userAgent,
+    // stringify logs upront because it's an array...
+    logs: JSON.stringify(appLogsArr)
+  })], { type: 'application/json' })
+
+  if (window.navigator.msSaveOrOpenBlob) {
+    // IE10+ and Edge
+    console.log('download using MS API')
+    window.navigator.msSaveOrOpenBlob(file, filename)
+  } else {
+    console.log('download using URL obj')
+
+    const url = URL.createObjectURL(file)
+    elLink.href = url
+    elLink.download = filename
+    elLink.click()
+    setTimeout(() => {
+      elLink.href = '#'
+      window.URL.revokeObjectURL(url)
+    }, 0)
+  }
+}
+
+export function clearLogs () {
+  let i = localStorage.length
+  while (i--) {
+    const key = localStorage.key(i)
+    if (/giConsole/.test(key)) {
+      localStorage.removeItem(key)
+    }
+  }
+  lastEntry = ''
+  entriesCount = 0
```

Comment:
With the modified version of `verifyLogsConfigs()`, we can just call `verifyLogsConfigs()` here instead and it should update both these variables and all of the localStorage values.

Reply:
I moved these two lines to `verifyLogsConfig`.

Reply:
Wait, but these two lines are still here, and `verifyLogsConfigs()` hasn't been called..? We need a call to `verifyLogsConfigs()` here to reset everything and ensure all variables are correctly set in both memory and localStorage

Reply:
Hum... my bad on the previous comment. I was talking about `captureLogsStart`.

Anyway... this is `clearLogs`. Its only purpose is to clear (delete) all the logs. Why would we need to `verifyLogsConfigs()` here?

Reply:
Because it resets the `localStorage` values as well.

Reply:
Yes, it clears the localStorage as expected. We only need to `verifyLogsConfig` (a.k.a reset the LS to its initial values) once we login back again.

Reply:
That doesn't make sense to me. `clearLogs()` can be called in the middle of a logged in session from `verifyLogsSize()`, or via some other means that might be added later. So it needs to reset everything back to normal by calling `verifyLogsConfigs()` here.

Reply:
One thing is clearLogs, another is resetLogs. Here, we want to clear them, delete them for good. If we call verifyLogsConfigs it will set again `limit`, `markers` and `markerNth` and that doesn't make sense and it isn't needed.

Reply:
It also resets `giLSset('count', entriesCount)`, and that is very much needed.

Reply:
Not, it doesn't and it isn't needed. `entriesCount` is only set at `captureLog()` and deleted at `clearLogs`.

Reply:
Well, it's true that it's deleted here, but it should be reset as well.

If `clearLogs()` is called, doesn't it make sense to make sure that valid values are restored in LS for `limit`, `markers`, `markerNth`, and everything else?

I can see how the code might work *at this time* without doing so, but if anything is slightly adjusted, things can break. Adding `verifyLogsConfigs()` to clear *and* reset everything seems to make sense to me to prevent fragility in the code.

Reply:
Let me give you a concrete example: say in the future we receive a PR that clears the logs using a different method, e.g. by looping through the logs and deleting them instead of looping through localStorage.

Then this code will suddenly break because `markerNth` wasn't restored.

Reply:
I think we should delete the logs and don't restore them so we have a brand new fresh start as if it was a completely new user, (where nothing is set yet). So it should always work even if anything is slightly adjusted, for a brand new user and for a user that did logout previously.

Reply:
That would normally make perfect sense, if you weren't already in the middle of being logged in, which is possible since `clearLogs()` is called from `verifyLogsSize()`.

So to prevent things from breaking, since `clearLogs()` does not log us out, we need to restore everything as if we were logging in again.

Reply:
Okay, so `clearLogs` and `resetLogs` are needed, each one for each case. pushed!